Extract.hs

```diff
@@ -0,0 +1,28 @@
+import System.IO
+import System.Environment
+import Data.List
+
+main = do
+    args <- getArgs
+    header <- return( args !! 0)
+    file <- return( args !! 1)
+    -- putStrLn ("Searching for '" ++ header ++ "' in " ++ file)
+    withFile file ReadMode (\handle -> do
+        contents <- hGetContents handle
+        putStr $ processText header contents)
+
+processText :: String -> String -> String
+processText tgt input =
+    let removed = dropWhile (\x -> not (matchesHeader tgt x)) $ lines input
```

Comment:
You could *probably* do
```suggestion
    let removed = dropWhile (not.(matchesHeader tgt)) $ lines input
```

to curry the `matchesHeader` function and chain `not` onto it.